src/TuDa.CIMS.Shared/Dtos/PaginatedResponse.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Microsoft.EntityFrameworkCore;
+
+namespace TuDa.CIMS.Shared.Dtos;
+
+/// <summary>
+/// A class that represents a paginated response.
+/// </summary>
+/// <typeparam name="T"></typeparam>
+public class PaginatedResponse<T> : List<T>
+{
+    public PaginatedResponse(IEnumerable<T> currentPage, int count, int pageNumber, int pageSize)
+    {
+        TotalCount = count;
+        PageSize = pageSize;
+        CurrentPage = pageNumber;
+        TotalPages = (int)Math.Ceiling(count / (double)pageSize);
+        AddRange(currentPage);
+    }
+
+    /// <summary>
+    /// The current page of the paginated response.
+    /// </summary>
+    public int CurrentPage { get; set; }
+
+    /// <summary>
+    /// The total number of pages in the paginated response.
+    /// </summary>
+    public int TotalPages { get; set; }
+
+    /// <summary>
+    /// The size of the page in the paginated response.
+    /// </summary>
+    public int PageSize { get; set; }
+
+    /// <summary>
+    /// The total number of items in the paginated response.
+    /// </summary>
+    public int TotalCount { get; set; }
+
+    /// <summary>
+    /// method to create a paginated response.
+    /// </summary>
+    /// <param name="source"></param>
+    /// <param name="pageNumber"></param>
+    /// <param name="pageSize"></param>
+    /// <returns></returns>
+    public static async Task<PaginatedResponse<T>> CreateAsync(IQueryable<T> source, int pageNumber, int pageSize)
+    {
+        var count = await source.CountAsync();
+        var items = await source.Skip((pageNumber - 1) * pageSize).Take(pageSize).ToListAsync();
```

Comment:
🌱 For slightly better performance, you could parallelize this. I would look something like this:
```suggestion
        var countTask = source.CountAsync();
        var itemsTask = source.Skip((pageNumber - 1) * pageSize).Take(pageSize).ToListAsync();

        Task.WaitAll(countTask, itemsTask);
        
        return new PaginatedResponse<T>(await itemsTask, await countTask, pageNumber, pageSize);
```